src/daemon/daemon.cpp

```diff
@@ -1465,6 +1465,25 @@ mp::Daemon::~Daemon()
     mp::top_catch_all(category, [this] {
         MP_SETTINGS.unregister_handler(instance_mod_handler);
         MP_SETTINGS.unregister_handler(snapshot_mod_handler);
+
+        /**
+         * Wait until all futures are finished, so there will
+         * be no outstanding requests left behind. Otherwise, the
+         * futures might be interrupted halfway and the actions
+         * associated with the futures are not going to be
+         * executed.
+         */
+        for (auto& [_, watcher] : async_future_watchers)
+        {
+            watcher->waitForFinished();
```

Comment:
This line is not always going to be hit by the unit tests since it depends on the timing.